subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java

```diff
@@ -240,6 +242,36 @@ public void setIgnoreFailures(boolean ignoreFailures) {
     }
 
     /**
+     * The maximum number of errors that are tolerated before breaking the build
+     * or setting the failure property.
+     *
+     * @return the maximum number of errors allowed
+     */
+    @Console
```

Comment:
These properties should be used for `UP-TO-DATE` checking because they affect the build outcome and so should be annotated `@Input`